src/codec.ts

```diff
@@ -30,6 +30,40 @@ export interface Field {
   value: Value;
 }
 
+export interface IProtoMessageParams {
+  // Supports proto message serialized binary data as a `Buffer` or pass a
+  // message object created using the functions generated by protobufjs-cli.
+  value: object;
+  // Fully qualified name of the proto representing the message definition
+  fullName: string;
+  /**
+   * Provide a First Class function that includes nested functions named
+   * "encode" for serialization and "decode" for deserialization of the proto
+   * message. The function should be sourced from the JS file generated by
+   * protobufjs-cli for the proto message.
+   */
+  messageFunction?: Function;
+}
+
+export interface IProtoEnumParams {
+  // Supports proto enum integer constant or pass the enum string
+  // the functions generated by protobufjs-cli.
+  value: string | number;
+  // Fully qualified name of the proto representing the enum definition
+  fullName: string;
+  /**
+   * An object containing enum string to id mapping.
+   * @example: { POP: 0, JAZZ: 1, FOLK: 2, ROCK: 3 }
+   *
+   * The object should be sourced from the JS file generated by
+   * protobufjs-cli for the proto message. Additionally, please review the
+   * sample at {@link www.samples.com} for guidance.
+   *
+   * ToDo: Update the link
```

Comment:
Remove this todo

Reply:
Removed the TODO and link as we don't follow this for existing methods